lib/util.js

```diff
@@ -208,6 +210,76 @@ const util = {
     fs.copySync(srcDir, dstDir)
   },
 
+  // In linux, WidevineCdm library is bundled instead of installing it by component
+  // updating during the runtime. In linux, we can't load cdm library during the runtime
+  // because processes except browser process are forked from zygote process not created from
+  // brave.exe binary. So, cdm should be loaded(not initialized) into zygote process space
+  // before zygote enters the sandbox.
+  // This method places WidevineCdm lib and header file into proper place before building.
+  prepareWidevineCdmLibIfNeeded: () => {
+    const widevineDir = path.join(config.srcDir, 'third_party', 'widevine', 'cdm', 'linux', 'x64')
+    fs.ensureDirSync(widevineDir)
+
+    const widevineConfig = {
+      widevineDir,
+      configuredVersion: config.widevineVersion,
+      versionFilePath: path.join(widevineDir, 'version'),
+      widevineCdmHeaderFilePath: path.join(widevineDir, 'widevine_cdm_version.h'),
+      widevineCdmLibFilePath: path.join(widevineDir, 'libwidevinecdm.so')
+    }
+
+    // If version file isn't existed, do fresh install.
+    if (!fs.existsSync(widevineConfig.versionFilePath)) {
+      console.log("No widevine version file.")
+      util.doPrepareWidevineCdmLib(widevineConfig)
+      return
+    }
+
+    // If installed version is different with configured version, do fresh install.
+    const installedVersion = fs.readFileSync(widevineConfig.versionFilePath, 'utf8')
+    if (installedVersion !== widevineConfig.configuredVersion) {
+      console.log("Different widevine version is installed")
+      util.doPrepareWidevineCdmLib(widevineConfig)
+      return
+    }
+
+    // Double check necessary header and lib exists.
+    if (fs.existsSync(widevineConfig.widevineCdmHeaderFilePath) &&
+        fs.existsSync(widevineConfig.widevineCdmLibFilePath))
+      return;
+
+    util.doPrepareWidevineCdmLib(widevineConfig)
+  },
+
+  doPrepareWidevineCdmLib: (widevineConfig) => {
+    console.log('Install widevinecdm library')
+
+    // Create version file.
+    fs.writeFileSync(widevineConfig.versionFilePath, widevineConfig.configuredVersion)
+
+    // Crate header file.
+    const headerFileContent =
+`#ifndef WIDEVINE_CDM_VERSION_H_
+#define WIDEVINE_CDM_VERSION_H_
+#define WIDEVINE_CDM_VERSION_STRING \"${widevineConfig.configuredVersion}\"
+#endif  // WIDEVINE_CDM_VERSION_H_`;
+    fs.writeFileSync(widevineConfig.widevineCdmHeaderFilePath, headerFileContent)
+
+    // Synchronously download widevine pkg and extracts into cdm dir.
+    const widevineZipFileName = `${widevineConfig.configuredVersion}-linux-x64.zip`
+    const widevineUrl = `https://redirector.gvt1.com/edgedl/widevine-cdm/${widevineZipFileName}`
```

Comment:
I'm not sure if downloading a new binary each time (instead of the manual binary update) is OK from the security point of view, so maybe it is worth to show this to someone from DevOps or security teams.

Reply:
@bbondy  I think we also need to think about widevine download url could be exposed or not.

Reply:
Please post a security review for this PR entirely in brave/internal